### ISSUE TITLE
Enhancement: [ban-ts-comment] Autofix ts-ignore to ts-expect-error depending on options

### ISSUE THREAD

**Issue by NotWoods** [typescript-eslint/typescript-eslint#7843]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/ban-ts-comment/

### Description

A fairly common use-case for `ban-ts-comment` is to ban @ts-ignore in favor of @ts-expect-error.

However, if developers are unfamiliar with @ts-expect-error, the error message doesn't help them find out about it.

![](https://github.com/typescript-eslint/typescript-eslint/assets/1782266/b94c3cf9-2bb5-4ae2-a8bd-cb9e8a09f3f5)

The error message should recommend switching, and an autofix should be provided. If @ts-expect-error is also banned, the above original error message can be used instead.

I think it would be useful to autofix even if the rule is configured to require descriptions, as it still saves keystrokes. I don't know if autofixes are allowed to adjust code into a "fail" state however.

### Fail

```typescript
// @ts-ignore
```

### Pass

```typescript
// @ts-expect-error
// ^autofix
```

**Comment by Josh-Cena** [comment]
I'm +1 on improving the error message to potentially include `@ts-expect-error`, but autofix feels useless, partly because, as you mentioned, it may produce incorrect code, and partly because it doesn't change anything in practice otherwise—an unreasonably suppressed error stays suppressed. Maybe an editor suggestion fixer would help, though.

**Comment by bradzacher** [comment]
Yeah I don't think it's correct to autofix switch things. IMO it's very rare that you actually need an ignore comment and often (from my experience at scale) people are being lazy. With that in mind being short with the wording can push them to looking for better solutions - the right solutions.

For power users though the wording definitely is wrong. "I know I am using this ignore on purpose - I don't have another option - so why isn't it saying to use expect-error".

The choice is really whether or not you add more words to the message and potentially give people an easy out.

**Comment by NotWoods** [comment]
I agree that suggestion would be nicer than autofix 😁
